MaterialLib/MPL/MediumAuxiliaryFunction.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * File:   MediumAuxiliaryFunction.h
+ *
+ * Created on July 29, 2019, 2:48 PM
+ */
+
+#pragma once
+
+#include "MaterialLib/MPL/PropertyType.h"
+
+namespace MaterialPropertyLib
+{
+template <int GlobalDim>
+Eigen::Matrix<double, GlobalDim, GlobalDim> localTensor(
+    MaterialPropertyLib::PropertyDataType const& values)
+{
+    if (boost::get<double>(&values))
+    {
+        return Eigen::Matrix<double, GlobalDim, GlobalDim>::Identity() *
+               boost::get<double>(values);
+    }
+    if (boost::get<MaterialPropertyLib::Vector>(&values))
+    {
+        return Eigen::Map<Eigen::Matrix<double, GlobalDim, 1> const>(
+                   boost::get<MaterialPropertyLib::Vector>(values).data(),
+                   GlobalDim, 1)
+            .asDiagonal();
+    }
+    if (boost::get<MaterialPropertyLib::Tensor2d>(&values))
+    {
+        return Eigen::Map<Eigen::Matrix<double, GlobalDim, GlobalDim> const>(
+            boost::get<MaterialPropertyLib::Tensor2d>(values).data(), GlobalDim,
+            GlobalDim);
+    }
+    if (boost::get<MaterialPropertyLib::Tensor>(&values))
+    {
+        return Eigen::Map<Eigen::Matrix<double, GlobalDim, GlobalDim> const>(
+            boost::get<MaterialPropertyLib::Tensor>(values).data(), GlobalDim,
+            GlobalDim);
+    }
+    OGS_FATAL(
+        "The size of local tensor such intrinsic permeability, thermal "
+        "conductovity is neither one nor %d nor %d squared.",
+        GlobalDim, GlobalDim);
+}
+
+template <int GlobalDim>
```

Comment:
MPL is a general concept of properties. The calculation of the effective thermal conductivity is process specific, i.e., it is used only by some of the processes. For this reason I suggest to leave the function in its old place. By the way, the new function isn't used anywhere.

Reply:
The function will be used in TH related processes, and its first application will be in THM. After moved to ProcessLib/Utils, I would like to keep the function in this file (now renamed as FormTensor.h)